src/tss2-esys/esys_tcti_default.c

```diff
@@ -1,56 +1,214 @@
-#include "tss2_mu.h"
-#include "tss2_sys.h"
-#include "tss2_tcti_device.h"
-#include "tss2_tcti_mssim.h"
+/*******************************************************************************
+ * Copyright 2017-2018, Fraunhofer SIT sponsored by Infineon Technologies AG
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************/
+
+#include <stdlib.h>
 #include <errno.h>
-#include <stdio.h>
 #include <string.h>
 #include <inttypes.h>
+#ifndef NO_DL
+#include <dlfcn.h>
+#endif /* NO_DL */
 
-#ifndef ESYS_TCTI_DEFAULT_MODULE
-#define ESYS_TCTI_DEFAULT_MODULE Mssim
-#endif
-#ifndef ESYS_TCTI_DEFAULT_CONFIG
-#define ESYS_TCTI_DEFAULT_CONFIG tcp://127.0.0.1:2321
-#endif
+#include "tss2_tcti.h"
+#include "tss2_tcti_device.h"
+#include "tss2_tcti_mssim.h"
 
-#define _CONC(a,b,c) a ## b ## c
-#define _XCONC(a,b,c) _CONC(a,b,c)
-#define Tss2_Tcti_Default_Init _XCONC(Tss2_Tcti_, ESYS_TCTI_DEFAULT_MODULE, _Init)
+#define LOGMODULE esys
+#include "util/log.h"
 
 #define _STR(A) #A
 #define _XSTR(A) _STR(A)
 
-/*
- * Initialize a TCTI instance.
- * The caller is returned a TCTI context structure that is allocated by this
- * function. This structure must be freed by the caller.
- */
-TSS2_RC
-get_tcti_default(TSS2_TCTI_CONTEXT ** tcticontext)
+#define ARRAY_SIZE(X) (sizeof(X)/sizeof(X[0]))
+
+struct {
+    char *file;
```

Comment:
#ifndef NO_DL around file?

Reply:
@tstruk I think I don't know what you mean ?

Reply:
Just saying that we could also wrap the char *file member with #ifndef NO_DL:
```
struct {
    #ifndef NO_DL
    char *file;
    #endif
    TSS2_TCTI_INIT_FUNC init;
    TSS2_RC	+    char *conf;
    get_tcti_default(TSS2_TCTI_CONTEXT ** tcticontext)	+    char *description;
};
```

Reply:
ok, done...